src/ports/catalog/queries.ts

```diff
@@ -0,0 +1,355 @@
+    WHERE 
+        satsuma_subgraph_name = ${subgraphName}
+    `
+
+export function getOrderBy(filters: CatalogFilters) {
+  const { sortBy, sortDirection, onlyListing, isOnSale } = filters
+  const sortByParam = sortBy ?? CatalogSortBy.NEWEST
+  const sortDirectionParam = sortDirection ?? CatalogSortDirection.ASC
+
+  // When seeing "Not for sale", the only sort available is the Newest one
+  if (isOnSale === false) {
+    if (sortByParam === CatalogSortBy.NEWEST) {
+      return `ORDER BY items.first_listed_at ${sortDirectionParam}\n`
+    }
+    return ''
+  }
+
+  let sortByQuery:
+    | SQLStatement
+    | string = `ORDER BY items.first_listed_at ${sortDirectionParam}\n`
+  switch (sortByParam) {
+    case CatalogSortBy.NEWEST:
+      sortByQuery = `ORDER BY items.first_listed_at ${sortDirectionParam}\n`
+      break
+    case CatalogSortBy.MOST_EXPENSIVE:
+      sortByQuery = `ORDER BY max_price ${sortDirectionParam}\n`
+      break
+    case CatalogSortBy.RECENTLY_LISTED:
+      sortByQuery = onlyListing
+        ? `ORDER BY nfts.max_order_created_at ${sortDirectionParam}\n`
+        : ``
+      break
+    case CatalogSortBy.RECENTLY_SOLD:
+      sortByQuery = `ORDER BY sold_at ${sortDirectionParam}\n`
+      break
+    case CatalogSortBy.CHEAPEST:
+      sortByQuery = `ORDER BY min_price ${sortDirectionParam}\n`
+      break
+  }
+
+  return sortByQuery
+}
+
+export const addQuerySortAndPagination = (
+  query: SQLStatement,
+  filters: CatalogQueryFilters
+) => {
+  const { sortBy, sortDirection, limit, offset } = filters
+  if (sortBy && sortDirection) {
+    query.append(getOrderBy(filters)) // VER TODO
+  }
+
+  if (limit !== undefined && offset !== undefined) {
+    query.append(SQL`LIMIT ${limit} OFFSET ${offset}`)
+  }
+}
+
+export const getCategoryWhere = (
+  category: NFTCategory,
+  isWearableSmart?: boolean
+) => {
+  return category === NFTCategory.WEARABLE
+    ? isWearableSmart
+      ? SQL`items.item_type = '`
+          .append(FragmentItemType.SMART_WEARABLE_V1)
+          .append(SQL`'`)
+      : SQL`(items.item_type = '`
+          .append(FragmentItemType.WEARABLE_V1)
+          .append(SQL`' OR items.item_type = '`)
+          .append(FragmentItemType.WEARABLE_V2)
+          .append(SQL`' OR items.item_type = '`)
+          .append(FragmentItemType.SMART_WEARABLE_V1)
+          .append(SQL`')`)
+    : category === NFTCategory.EMOTE
+    ? SQL`items.item_type = '`.append(FragmentItemType.EMOTE_V1).append(SQL`'`)
+    : SQL``
+}
+
+export const getWearableCategoryWhere = (
+  wearableCategory: WearableCategory
+) => {
+  return WearableCategory.validate(wearableCategory)
+    ? SQL`metadata_wearable.category = '`
+        .append(wearableCategory)
+        .append(SQL`'`)
+    : SQL``
+}
+
+export const getEmoteCategoryWhere = (emoteCategory: EmoteCategory) => {
+  return EmoteCategory.validate(emoteCategory)
+    ? SQL`metadata_emote.category = '`.append(emoteCategory).append(SQL`'`)
+    : SQL``
+}
```

Comment:
I think `getWearableCategoryWhere` and `getEmoteCategoryWhere` have similar behavior. We could write a new function called `getCategoryWhere` or `getMetadataCategoryWhere` to generalize the duplicated logic. WDYT?

Reply:
I think the abstraction will be more complex than having 1 line duplicated. Specially the piece of validating the enum, if it recieves one enum or the other, we can't validate which is the one being called, it would requiere something like another parameter with a hardcoded string and having some ternaries for the validation and the SQL piece. I tried but it think that sometimes is better to have more code than having the wrong abstraction